src/batch_norm.cpp

```diff
@@ -565,19 +570,52 @@ void bnBwdTrainSelectSingle(const Handle& handle,
                             double epsilon,
                             ConstData_t savedMean,
                             ConstData_t savedInvVariance,
-                            float inhw)
+                            float nhw,
+                            unsigned int in_cstride,
+                            unsigned int in_nstride)
 {
 
     visit_float(dtype, [&](auto as_float) {
-        if(useSaved)
+        float inhw = float(1.0) / nhw;
+        if(variant == 4)
         {
-            handle.AddKernel(algo_name, network_config, program_name, kernel_name, vld, vgd, parms)(
-                x, dy, dx, bnScale, dScale, dBias, savedMean, savedInvVariance, as_float(inhw));
+            if(useSaved)
+            {
+                handle.AddKernel(
+                    algo_name, network_config, program_name, kernel_name, vld, vgd, parms)(
+                    x,
+                    dy,
+                    dx,
+                    bnScale,
+                    dScale,
+                    dBias,
+                    savedMean,
+                    savedInvVariance,
+                    nhw,
+                    in_cstride,
+                    in_nstride);
+            }
+            else
+            {
+                handle.AddKernel(
+                    algo_name, network_config, program_name, kernel_name, vld, vgd, parms)(
+                    x, dy, dx, bnScale, dScale, dBias, epsilon, nhw, in_cstride, in_nstride);
+            }
         }
         else
         {
-            handle.AddKernel(algo_name, network_config, program_name, kernel_name, vld, vgd, parms)(
-                x, dy, dx, bnScale, dScale, dBias, epsilon, as_float(inhw));
+            if(useSaved)
```

Comment:
is it not possible to club "useSaved" and not saved options into one single call?